ansibullbot/utils/file_tools.py

```diff
@@ -29,23 +28,14 @@ class FileIndexer(ModuleIndexer):
 
     files = []
 
-    def __init__(self, botmetafile=None, checkoutdir=None, repo=None):
-
-        if checkoutdir is None:
-            self.checkoutdir = '~/.ansibullbot/cache/ansible.files.checkout'
-        else:
-            self.checkoutdir = os.path.join(checkoutdir, 'ansible.files.checkout')
-        self.checkoutdir = os.path.expanduser(self.checkoutdir)
-
-        if repo:
-            self.REPO = 'https://github.com/{}'.format(repo)
-
+    def __init__(self, botmetafile=None, gitrepo=None):
         self.botmetafile = botmetafile
         self.botmeta = {}
         self.CMAP = {}
         self.FILEMAP = {}
         self.match_cache = {}
-        self.update(force=True)
+        self.gitrepo = gitrepo
```

Comment:
when `update` isn't called then `self.botmeta` isn't populated then this exception occurs:
```
  File "./triage_ansible.py", line 42, in <module>
    main()
  File "./triage_ansible.py", line 38, in main
    AnsibleTriage().start()
  File "ansibullbot/ansibullbot/triagers/defaulttriager.py", line 184, in start
    self.run()
  File "ansibullbot/ansibullbot/triagers/ansible.py", line 464, in run
    self.process(iw)
  File "ansibullbot/ansibullbot/triagers/ansible.py", line 1863, in process
    self.valid_labels
  File "ansibullbot/ansibullbot/triagers/plugins/component_matching.py", line 56, in get_component_match_facts
    CM_MATCHES = component_matcher.match(iw)
  File "ansibullbot/ansibullbot/utils/component_tools.py", line 288, in match
    files=iw.files
  File "ansibullbot/ansibullbot/utils/component_tools.py", line 344, in match_components
    component_matches.append(self.get_meta_for_file(fn))
  File "ansibullbot/ansibullbot/utils/component_tools.py", line 1023, in get_meta_for_file
    botmeta_entries = self.file_indexer._filenames_to_keys(filenames)
  File "ansibullbot/ansibullbot/utils/file_tools.py", line 125, in _filenames_to_keys
    if filen in self.botmeta['files']:
KeyError: 'files'
```

Reply:
Yeah, sorry, I noticed this just after I cc'ed you. Maybe I can consolidate botmeta file reading in this PR too...

Reply:
Note that `ModuleIndexer.update` calls both `get_file` and `parse_metadata` methods and `FileIndexer.parse_metadata` calls also `get_file`. Anyway both classes `ModuleIndexer`/`FileIndexer` need a cleanup/refactor.